api/src/auth/__tests__/auth.controller.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { AuthController } from '../auth.controller';
+import { AuthService } from '../auth.service';
+
+describe('AuthController', () => {
+  let controller: AuthController;
+  let authService: Partial<Record<keyof AuthService, jest.Mock>>;
+
+  beforeEach(async () => {
+    authService = {
+      login: jest.fn(),
+      register: jest.fn(),
+      validateUser: jest.fn(),
+    };
+
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [AuthController],
+      providers: [{ provide: AuthService, useValue: authService }],
+    }).compile();
+
+    controller = module.get<AuthController>(AuthController);
+  });
+
+  it('should login successfully', async () => {
+    authService.validateUser!.mockResolvedValue({ id: 1, username: 'test', email: 'test@test.com' });
+    authService.login!.mockReturnValue({ access_token: 'jwt-token' });
+
+    const result = await controller.login({ email: 'test@test.com', password: '123456' });
+    expect(result).toEqual({ access_token: 'jwt-token' });
+  });
+
+  it('should throw error if login fails', async () => {
```

Comment:
in the previous test it says "should login successfully" and now it will throw error but you are testing the same `controller.login` function - so what changed here for a positive and negative outcome?